infrastructure/microvm/cloudhypervisor/state.go

```diff
@@ -0,0 +1,116 @@
+package cloudhypervisor
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+
+	"github.com/spf13/afero"
+	"github.com/weaveworks-liquidmetal/flintlock/core/models"
+	"github.com/weaveworks-liquidmetal/flintlock/pkg/defaults"
+)
+
+type State interface {
+	Root() string
+
+	PID() (int, error)
+	PIDPath() string
+	SetPid(pid int) error
+
+	LogPath() string
+	StdoutPath() string
+	StderrPath() string
+	SockPath() string
+
+	//MetadataDir() string
+	CloudInitImage() string
+}
+
+func NewState(vmid models.VMID, stateDir string, fs afero.Fs) State {
+	return &fsState{
+		stateRoot: fmt.Sprintf("%s/%s", stateDir, vmid.String()),
+		fs:        fs,
+	}
+}
+
+type fsState struct {
+	stateRoot string
+	fs        afero.Fs
+}
+
+func (s *fsState) Root() string {
+	return s.stateRoot
+}
+
+func (s *fsState) PIDPath() string {
+	return fmt.Sprintf("%s/cloudhypervisor.pid", s.stateRoot)
```

Comment:
i'd like to see these file names be consts, but nbd right now

Reply:
Added constant